src/MaskingContainer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Dhii\Container;
+
+use Dhii\Container\Exception\NotFoundException;
+use Dhii\Container\Util\StringTranslatingTrait;
+use Psr\Container\ContainerInterface;
+use function array_key_exists;
+
+/**
+ * An implementation of a container that wraps around another to selectively expose or mask certain keys.
+ *
+ * @since [*next-version*]
+ */
+class MaskingContainer implements ContainerInterface
+{
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $inner;
+
+    /**
+     * @var bool[]
+     */
+    protected $mask;
+
+    /**
+     * @var bool
+     */
+    protected $defMask;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface $inner   The container whose entries to mask.
+     * @param bool[]             $mask    A mapping of keys to booleans, such that `true` exposes the mapped key and
+     *                                    `false` hides the mapped key.
+     * @param bool               $default The default mask value to use for keys that are not included in the mask.
+     */
+    public function __construct(ContainerInterface $inner, array $mask, $default = true)
+    {
+        $this->inner = $inner;
+        $this->mask = $mask;
+        $this->defMask = $default;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @since [*next-version*]
+     */
+    public function get($key)
+    {
+        if (!$this->isExposed($key)) {
+            throw new NotFoundException(
+                $this->__('Key "%1$s" was not found in the inner container or is not exposed', [$key]),
```

Comment:
IMO the message should be more explicit about the key not being exposed. Why should it mention the possibility of the key not being found?

Reply:
Probably a mistake from my end. The condition was likely checking for `has($key)`, but was later changed to `isExposed($key)` and the message incorrectly stayed the same.